fennel/internal_lib/to_proto/to_proto.py

```diff
@@ -134,17 +135,28 @@ def to_sync_request_proto(
             if offline_index:
                 offline_indices.append(offline_index)
 
-            res = sources_from_ds(obj, obj.timestamp_field, tier)
-            if res is None:
-                continue
-            (ext_db, s) = res
-            conn_sources.append(s)
-            # dedup external dbs by the name
-            # TODO(mohit): Also validate that if the name is the same, there should
-            # be no difference in the other fields
-            if ext_db.name not in external_dbs_by_name:
-                external_dbs_by_name[ext_db.name] = ext_db
-                external_dbs.append(ext_db)
+            sources = sources_from_ds(obj, obj.timestamp_field, tier)
```

Comment:
can we make sources_from_ds identical to sinks_from_ds? i.e. not return an Optional[] but actually list as you have done. It would make code flow a bit easier to follow (e.g. how it is for sinks right now)

Any reason why these two have different return types (they are extracted using similar information and the only diff is the source/sink).

Reply:
Understood